AmazonChimeSDKDemo/AmazonChimeSDKDemo/LiveTranscriptionOptionsViewController.swift

```diff
@@ -139,31 +223,150 @@ class LiveTranscriptionOptionsViewController: UIViewController, UITextFieldDeleg
+        
+        languageDetectionViewController?.cancellationHandler = { numberOfselectedLanguages in
+            self.enableAutomaticLanguageDetectionSwitch.isOn = numberOfselectedLanguages! > 1
+        }
+        
+        languageDetectionViewController?.selectedLanguageOptionsHandler = { selectedLanguages in
+            self.selectedLanguagesDetectionOptions = selectedLanguages!
+        }
+        
+        languageDetectionViewController?.preferredLanguageOptionHandler = { preferredLanguage in
+            self.preferredLanguagesDetectionOptions = preferredLanguage!
+        }
+        
+        self.languageHandler = { languageList in
+            self.languageDetectionViewController?.languages = languageList!
+        }
+        
+        self.languagesDictHandler = { mappings in
+            self.languageDetectionViewController?.languagesDict = mappings!
+        }
     }
     
     @IBAction func startTranscriptionButton(_ sender: UIButton) {
-        let url = self.meetingEndpointUrl.hasSuffix("/") ? self.meetingEndpointUrl : "\(self.meetingEndpointUrl)/"
+        var url = AppConfiguration.url
+        url = url.hasSuffix("/") ? url : "\(url)/"
+        
+        let contentIdentificationEnabled = piiContentIdentificationSelected != ""
+        let contentRedactionEnabled = piiContentRedactionSelected != ""
+        
+        var identifyLanguage: Bool?
+        var enablePartialResultsStabilization: Bool?
+        var contentIdentificationType: String?
+        var contentRedactionType: String?
+        var partialResultsStabilty: String?
+        var piiEntityTypes: String?
+        var languageModelName: String?
+        var languageOptions: String?
+        var preferredLanguage: String?
+        
+        if contentIdentificationEnabled {
+            if engineSelected == "transcribe_medical" {
+                contentIdentificationType = "PHI"
+            } else {
+                piiEntityTypes = piiContentIdentificationSelected
+                contentIdentificationType = "PII"
+            }
+        }
+        
+        if contentRedactionEnabled {
+            if engineSelected == "transcribe_medical" {
+                contentRedactionType = "PHI"
+            } else {
+                piiEntityTypes = piiContentRedactionSelected
+                contentRedactionType = "PII"
+            }
+        }
+        
+        if partialResultsStabilizationSelected > 0 {
+            enablePartialResultsStabilization = true
+            if partialResultsStabilizationSelected == 1 {
+                partialResultsStabilty = ""
+            } else {
+                partialResultsStabilty = partialResultsStabilizationsList[partialResultsStabilizationSelected].lowercased()
+            }
+        }
+        
+        if enableCustomLangugeModelTextFieldSwitch.isOn {
+            if customLanguageModelTextField.text!.isEmpty {
```

Comment:
avoid `!`

Reply:
updated all unsure occurrences of `!` to `?? <default value>`